### PR TITLE
avoid trying to pass null into useTube on reconnect.

### PR DESCRIPTION
**Pull request by DavidGoodwin**

### CODE REVIEW

File: src/Pheanstalk.php [hunk 1]
```diff
@@ -383,7 +383,7 @@ private function reconnect()
     {
         $this->connection->disconnect();
 
-        if ($this->using != PheanstalkInterface::DEFAULT_TUBE) {
+        if ($this->using !== null && $this->using != PheanstalkInterface::DEFAULT_TUBE) {
```
Comment (SamMousa) [inline src/Pheanstalk.php:386]: Hmm, this happens only if an exception is thrown during the `useTube` command which is called after a reconnect.
Instead it makes more sense to just dispatch the command remove the hacky state chage.

File: src/Pheanstalk.php [hunk 2]
```diff
@@ -383,7 +383,7 @@ private function reconnect()
...
             $tube = $this->using;
             $this->using = null;
             $this->useTube($tube);
```
Comment (SamMousa) [inline src/Pheanstalk.php:389]:
```suggestion
            $this->dispatch(new Command\UseCommand($this->using));
```

File: src/Pheanstalk.php (hunk 1 above)
Comment (SamMousa) [inline src/Pheanstalk.php:386]: This reverts your change and adds strict comparison.
```suggestion
        if ($this->using !== PheanstalkInterface::DEFAULT_TUBE) {
```

**Comment by DavidGoodwin** [comment]
Thanks for the feedback.

**Pull request merged**